tests/bor/bor_test.go

```diff
@@ -268,3 +272,159 @@ func getSampleEventRecord(t *testing.T) *bor.EventRecordWithTime {
 	_eventRecords[0].Time = time.Unix(1, 0)
 	return _eventRecords[0]
 }
+
+// TestEIP1559Transition tests the following:
+//
+// 1. A transaction whose gasFeeCap is greater than the baseFee is valid.
+// 2. Gas accounting for access lists on EIP-1559 transactions is correct.
+// 3. Only the transaction's tip will be received by the coinbase.
+// 4. The transaction sender pays for both the tip and baseFee.
+// 5. The coinbase receives only the partially realized tip when
+//    gasFeeCap - gasTipCap < baseFee.
+// 6. Legacy transaction behave as expected (e.g. gasPrice = gasFeeCap = gasTipCap).
+func TestEIP1559Transition(t *testing.T) {
```

Comment:
@ferranbt please take a look at the test case
1 - using engine as ethash
2 - changed ethash to bor config
3 - not sure but ethash seems to be adding some block rewards to the coinbase during its consensus. so i have kept it as it is. let me know if this makes sense or not